### PR TITLE
chore(ci): Adding separate e2e smoke tests on PR check

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Reenabled PR check with e2e smoke tests in it. `test:e2e` npm task is renamed to `test:e2e:smoke`.
### What issues does this PR fix or reference?
#2707

### CODE REVIEW

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -172,7 +172,7 @@ jobs:
 
   lint-format-unit:
     name: linter, formatters and unit tests
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: this change is not related to the issue (so should be a separate PR)

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -203,6 +203,9 @@ jobs:
       - name: Run unit tests
         run: yarn test:unit
 
+      - name: Run E2E smoke tests
+        run: yarn test:e2e:smoke
+
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: AFAIK we can't put it the smoke e2e as a step

it needs to be a separate job so if it fails (as it seems it is not stable we can still merge PR)

Comment (odockal) [reply .github/workflows/pr-check.yaml]: I am getting lost. That is exactly what I was trying to achieve by https://github.com/containers/podman-desktop/pull/2774 and there was disagreement over having e2e not a part of PR check. What I am missing?

Comment (benoitf) [reply .github/workflows/pr-check.yaml]: @odockal disagreement is about when to run it.

We said we wanted to be part of a PR check, not when we push to the main branch.

your PR https://github.com/containers/podman-desktop/pull/2774 trigger the workflow when there is a new commit in main branch
it should be a job triggered when user file a new PR

Comment (benoitf) [reply .github/workflows/pr-check.yaml]: so, in short, just add a new job at the end of pr-check.yaml file (a job, not a step)

Comment (odockal) [reply .github/workflows/pr-check.yaml]: I see! thanks for explanation.

**Review by benoitf — APPROVED** [review]

File: .github/workflows/pr-check.yaml [hunk 3]
```diff
@@ -214,3 +214,36 @@ jobs:
         with:
           name: tests
           path: ./tests/output/
+
+  e2e-tests:
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]:
```suggestion
  smoke-e2e-tests:
```
